### ISSUE TITLE
Whitespace in program arguments is parsed incorrectly

### ISSUE THREAD

**Issue by Gekkio** [ferrous-systems/flamegraph#23]
If there's any whitespace in the arguments intended for the profiled program, it looks like those arguments are incorrectly split into multiple individual parts.

src/main.rs:

```rust
fn main() {
    println!("{:?}", std::env::args().skip(1).collect::<Vec<_>>());
}
```

Running normally:

```shell
$ cargo run 'space cadet'
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
     Running `target/debug/foo 'space cadet'`
["space cadet"]
```

Running with cargo-flamegraph:

```shell
$ cargo flamegraph 'space cadet'
    Finished release [optimized + debuginfo] target(s) in 0.00s
["space", "cadet"]
[ perf record: Woken up 1 times to write data ]
[ perf record: Captured and wrote 0,018 MB perf.data (5 samples) ]
writing flamegraph to "flamegraph.svg"
```

Note how the program is incorrectly getting two arguments instead of just one. In practice this makes it impossible to pass any arguments with whitespace to the program.

```shell
$ cargo-flamegraph --version
cargo-flamegraph 0.1.9
```

**Comment by spacejam** [comment]
Hmm indeed, the implementation is currently quite simple around this, and it also behaves differently on linux than from one of the dtrace platforms, effectively `$*` vs `$@` from bash.

I see one relatively straightforward workaround being to use `flamegraph` instead of `cargo-flamegraph` and change the way that `flamegraph` passes arguments through to the binary under test, relying directly on `std::env::args` instead of the clap passthrough. I'll let this stew for a little, as maybe there is an even simpler way to avoid this issue.

**Comment by rcoh** [comment]
How does normal `cargo run` handle it? The most confusing part is that a command that works with `cargo run` behaves differently under `cargo flamegraph`

**Comment by kylewlacy** [comment]
My guess is that the problem is that extra arguments get joined here with spaces: https://github.com/ferrous-systems/flamegraph/blob/cd5855f08ea12143272c276ed477a7c6903eae67/src/bin/cargo-flamegraph.rs#L270

Then split again here: https://github.com/ferrous-systems/flamegraph/blob/e83bcc9cad40df0c22979cc52145d77f878880f2/src/lib.rs#L50

I believe `workload` should be a `Vec<String>` (or `Vec<OsString>` maybe?) in order to preserve what the input arguments actually are.

**Comment by Ralith** [comment]
This is fixed by #51. Clap is doing the right thing here, it was just the flamegraph code mangling the result.

**Comment by jonas-schievink** [comment]
#51 has been merged.
